Import location data for historical workshops

pbanaszkiewicz opened the pull request:
The data is gathered from swcarpentry/site archives
(`config/archive.yml`).

This new management command also updates 'contact' field.

The command was needed because I introduced location fields to the AMY in v0.7, but no-one should update them manually…

gvwilson commented on workshops/management/commands/import_archived_workshops.py:
```diff
@@ -0,0 +1,77 @@
+                country=D.get('country', '').upper(),
+                venue=D.get('venue'), address=D.get('address'),
+                latitude=(D.get('latlng') or ',').split(',')[0],
+                longitude=(D.get('latlng') or ',').split(',')[1],
```
Why not `D.get('latlng', ',')` ?

pbanaszkiewicz replied:
Because empty `latlng` in workshop file will result in `None` (ie. `.get('latlng', ',')` will yield `None`, not `','`).

Similar trick should be used for `country`, I think…

gvwilson commented:
I have failing tests with this:

~~~
FAIL: test_match_by_country (workshops.test.test_instructors.TestLocateInstructors)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/gvwilson/s/amy/workshops/test/test_instructors.py", line 55, in test_match_by_country
    assert "Hermione Granger" in content
AssertionError

======================================================================
FAIL: test_match_by_multiple_countries (workshops.test.test_instructors.TestLocateInstructors)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/gvwilson/s/amy/workshops/test/test_instructors.py", line 66, in test_match_by_multiple_countries
    assert "Hermione Granger" in content
AssertionError

======================================================================
FAIL: test_match_gender (workshops.test.test_instructors.TestLocateInstructors)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/gvwilson/s/amy/workshops/test/test_instructors.py", line 77, in test_match_gender
    assert "Hermione Granger" in content
AssertionError

======================================================================
FAIL: test_match_instructors_on_one_skill (workshops.test.test_instructors.TestLocateInstructors)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/gvwilson/s/amy/workshops/test/test_instructors.py", line 31, in test_match_instructors_on_one_skill
    assert "Hermione Granger" in content
AssertionError

======================================================================
FAIL: test_match_instructors_on_two_skills (workshops.test.test_instructors.TestLocateInstructors)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/gvwilson/s/amy/workshops/test/test_instructors.py", line 44, in test_match_instructors_on_two_skills
    assert "Hermione Granger" in content
AssertionError

======================================================================
FAIL: test_non_instructors_not_returned_by_search (workshops.test.test_instructors.TestLocateInstructors)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/gvwilson/s/amy/workshops/test/test_instructors.py", line 19, in test_non_instructors_not_returned_by_search
    assert "Hermione Granger" in content
AssertionError
~~~

gvwilson commented:
Running this with the current `archived.yml` yields:

~~~
Historical event 2015-03-05-uvasom does not exist in the database
Historical event 2015-06-15-cambridge does not exist in the database
Historical event 2015-06-21-UH-SWC does not exist in the database
Historical event 2015-06-21-s2i2 does not exist in the database
Historical event 2015-06-25-notre-dame does not exist in the database
Historical event 2015-03-20-UNL does not exist in the database
Historical event 2015-02-25 does not exist in the database
Historical event 2015-06-18-ucl-swcarpentry does not exist in the database
~~~

which isn't surprising - AMY is probably behind on workshops.  Can you please modify it to also report what changes it made to workshop records it modifies if a `-v` flag is provided?  E.g., tell me that you've changed the date, added a lat/long, modified the list of instructors, etc?  That'll help me do spot checks.

rgaiacs commented:
@pbanaszkiewicz Can we have something that run daily and email coordinators about workshops that change the information at their homepage? Or we can add one webhook into workshops repositories (I need to take a look at GitHub' API to see if this is possible).

pbanaszkiewicz commented:
@gvwilson:
> I have failing tests with this:

The tests aren't failing for me nor for Travis-CI. Not sure what code did you run.

@gvwilson:
> which isn't surprising - AMY is probably behind on workshops

At least one of these workshops exists in AMY, but is named differently (`2015-06-18-ucl` instead of `2015-06-18-ucl-swcarpentry`).

> Can you please modify it to also report what changes it made to workshop records it modifies if a -v flag is provided?

Sure :)

>  E.g., tell me that you've changed the date, added a lat/long, modified the list of instructors, etc? That'll help me do spot checks.

Currently it's only for location and contact info… do you want it to be a broader tool?

@rgaiacs:
> Can we have something that run daily and email coordinators about workshops that change the information at their homepage? Or we can add one webhook into workshops repositories (I need to take a look at GitHub' API to see if this is possible).

Not sure if we're capable of doing something like this…

gvwilson commented:
The plan is that we'll use this once to import data from site into AMY, 
and thereafter export data from AMY to site every time we build the site 
(getting rid of config/workshops.yml and config/archived.yml).

pbanaszkiewicz commented:
Hi,

On Mon, Aug 10, 2015 at 6:36 PM, Greg Wilson <notifications@github.com>
wrote:

> The plan is that we'll use this once to import data from site into AMY,
> and thereafter export data from AMY to site every time we build the site
> (getting rid of config/workshops.yml and config/archived.yml).


Okay, but what data should we import into the AMY? Location / contact is
obvious (because missing). What else?

The "export from AMY" will be done via REST API (see
https://github.com/swcarpentry/amy/pull/459).

Cheers,
Piotr


>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/swcarpentry/amy/pull/462#issuecomment-129520331>.
>

pbanaszkiewicz commented:
@gvwilson how do you like the changes? Can I proceed with this PR or do you want me to implement importing more data than contact/location (https://github.com/swcarpentry/amy/pull/462#issuecomment-129486836)?
